### PR TITLE
TCVP-2136 mapping for dispute update request

### PR DESCRIPTION
**Pull request by karahanb**
# Description

This PR includes the following proposed change(s):

- [TCVP-2136](https://justice.gov.bc.ca/jira/browse/TCVP-2136)
- Added mapper for `DisputeUpdateRequest` and updated occm-openapi-spec in order to map ORDS data to Oracle Data API disputeUpdateRequest model and vice versa.
- Updated Oracle Data API OpenAPI spec client and refactored `DisputantUpdateRequest` into `DisputeUpdateRequest`.
- Added JUnit tests for the mapper.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

mvn verify

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by karahanb** [comment]
@lfdgov Can you please confirm whether the renaming `DisputantUpdateRequest` to `DisputeUpdateRequest` will break frontend or not? If there is a dependency on that naming we should update front-end as well.

File: src/backend/TrafficCourts/Workflow.Service/Consumers/DisputantUpdateRequestRejectedConsumer.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 using TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0;
 using TrafficCourts.Messaging.MessageContracts;
 using TrafficCourts.Workflow.Service.Services;
-using DisputantUpdateRequest = TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0.DisputantUpdateRequest;
+using DisputantUpdateRequest = TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0.DisputeUpdateRequest;
```
Comment (KDNeufeld) [inline src/backend/TrafficCourts/Workflow.Service/Consumers/DisputantUpdateRequestRejectedConsumer.cs:6]: If you're renaming DisputantUpdateRequest to DisputeUpdateRequest, should this `using` statement be updated as well to be consistent?

```suggestion
using DisputeUpdateRequest = TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0.DisputeUpdateRequest;
```

File: src/backend/TrafficCourts/Workflow.Service/Consumers/DisputantUpdateRequestAcceptedConsumer.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 using TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0;
 using TrafficCourts.Messaging.MessageContracts;
 using TrafficCourts.Workflow.Service.Services;
-using DisputantUpdateRequest = TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0.DisputantUpdateRequest;
+using DisputeUpdateRequest = TrafficCourts.Common.OpenAPIs.OracleDataApi.v1_0.DisputeUpdateRequest;
 
 namespace TrafficCourts.Workflow.Service.Consumers;
```
Comment (KDNeufeld) [inline src/backend/TrafficCourts/Workflow.Service/Consumers/DisputantUpdateRequestAcceptedConsumer.cs:9]: This class DisputantUpdateRequestAcceptedConsumer should be updated to DisputeUpdateRequestAcceptedConsumer to be consistent

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/DisputeUpdateRequestMapper.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+	@Named("mapDisputeUpdateStatTypeCd")
+	default String mapDisputeUpdateStatTypeCd(DisputeUpdateRequestStatus status) {
+		return status.getShortName();
+	}
+
+	@Named("mapDisputeUpdateReqTypeCd")
+	default String mapDisputeUpdateReqTypeCd(DisputeUpdateRequestType status) {
+		return status.getShortName();
+	}
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/DisputeUpdateRequestMapper.java:76]: Both of these code mappers can be simplified by just using the interface.


```
	@Named("mapShortNamedEnum")
	default String mapShortNamedEnum(ShortNamedEnum code) {
		return code.getShortName();
	}
```

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/Auditable.java [hunk 4]
```diff
@@ -26,8 +26,8 @@
  */
 @TypeDefs({
 	@TypeDef(name = "contactType", defaultForType = ContactType.class, typeClass = ShortNamedEnumType.class),
-	@TypeDef(name = "disputantUpdateStatus", defaultForType = DisputantUpdateRequestStatus.class, typeClass = ShortNamedEnumType.class),
-	@TypeDef(name = "disputantUpdateType", defaultForType = DisputantUpdateRequestType.class, typeClass = ShortNamedEnumType.class),
+	@TypeDef(name = "disputantUpdateStatus", defaultForType = DisputeUpdateRequestStatus.class, typeClass = ShortNamedEnumType.class),
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/Auditable.java:29]:
```suggestion
	@TypeDef(name = "disputeUpdateRequestStatus", defaultForType = DisputeUpdateRequestStatus.class, typeClass = ShortNamedEnumType.class),

```

**Review by KDNeufeld — APPROVED** [review]
Great work. 
As Lorraine mentioned, confirm the UI is not affected by the refactoring name change.
There are still quite a few references to the word "DisputantUpdateRequest" in the .NET and oracle-data-api code bases. If we're going to refactor the name, lets go all the way.

**Comment by karahanb** [comment]
> Great work. As Lorraine mentioned, confirm the UI is not affected by the refactoring name change. There are still quite a few references to the word "DisputantUpdateRequest" in the .NET and oracle-data-api code bases. If we're going to refactor the name, lets go all the way.

Thanks for catching those remaining "DisputantUpdateRequest" references. I fixed all the ones that were causing errors but should have done some more extensive search. I will fix the remaining ones. Actually I asked Lorraine to confirm the UI is not affected since I'm not that familiar with frontend code but I will check that as well.

**Review by KDNeufeld — APPROVED** [review]
Great work! Refactoring looks good.

**Review by lfdgov — CHANGES_REQUESTED** [review]
update apis in client side source code

File: src/frontend/citizen-portal/package.json [hunk 5]
```diff
@@ -64,7 +64,7 @@
     "@angular-eslint/eslint-plugin-template": "14.4.0",
     "@angular-eslint/schematics": "14.4.0",
     "@angular-eslint/template-parser": "14.4.0",
-    "@angular/cli": "^14.2.10",
```
Comment (lfdgov) [inline src/frontend/citizen-portal/package.json:67]: change this back to 14.2.10 in package.json and then do yarn install

**Review by lfdgov — CHANGES_REQUESTED** [review]

**Review by lfdgov — APPROVED** [review]

**Pull request merged**